Mathlib/RingTheory/PowerSeries/Basic.lean

```diff
@@ -391,6 +404,18 @@ theorem coeff_X_pow_mul (p : R⟦X⟧) (n d : ℕ) :
   · rw [add_comm]
     exact fun h1 => (h1 (mem_antidiagonal.2 rfl)).elim
 
+theorem mul_X_pow_cancel {k : ℕ} {φ ψ : R⟦X⟧} (h : φ * X ^ k = ψ * X ^ k) :
+    φ = ψ := by
+  rw [PowerSeries.ext_iff] at h ⊢
+  intro n
+  simpa using h (n + k)
+
+theorem X_pow_mul_cancel {k : ℕ} {φ ψ : R⟦X⟧} (h : X ^ k * φ = X ^ k * ψ) :
+    φ = ψ := by
+  rw [PowerSeries.ext_iff] at h ⊢
+  intro n
+  simpa using h (n + k)
```

Comment:
Given these two lemma, are the ones for k=1 (`mul_X_cancel` and `X_mul_cancel`) really needed? They follow from these with a `simp`.

Reply:
I understand but if you want to simplify by `X`, do you want to first rewrite `X` as `X ^ 1`?
The proof is short, I'm not even sure it is useful to deduce the one for `X` from this one.